Content.Server/Weapons/Ranged/Systems/GunSystem.cs

```diff
@@ -133,8 +135,8 @@ private void OnBallisticPrice(EntityUid uid, BallisticAmmoProviderComponent comp
                 case CartridgeAmmoComponent cartridge:
                     if (!cartridge.Spent)
                     {
-                        if (gun.CompatibleAmmo != null &&
-                            !gun.CompatibleAmmo.Exists(ammoAllowed => ammoAllowed.Equals(cartridge.Prototype))
```

Comment:
I don't know who added this but it shouldn't be on guncomponent.

Reply:
https://github.com/space-wizards/space-station-14/pull/24679 so if you want this it will need re-implementing to not be hardcoded into guns.

Reply:
Where would you put it? It makes sense on guns to me, the gun is the thing firing the projectile after all.

Reply:
raise an event instead of hardcoding a feature for a *single* item into it

Reply:
> raise an event instead of hardcoding a feature for a _single_ item into it

Me when I start hardcoding in clumsy and fire rates every other event. Smug literally just refactored it to use events.

> Where would you put it? It makes sense on guns to me, the gun is the thing firing the projectile after all.

Use an event.

Reply:
okay, but how would I listen for the event from a projectile that isn't initialized yet?

Reply:
its not on a projectile its on the cartridge item

Reply:
Hmm, that might work. I think the cartridge is always initialized in the chamber.

Reply:
Oh I see what you meant. It was reading the projectile prototype off the cartridge datafield. Okay, that makes this a lot easier.